src/main/java/com/google/devtools/build/lib/bazel/repository/GitCloner.java

```diff
@@ -181,6 +189,15 @@ public static SkyValue clone(Rule rule, Path outputDirectory, EventHandler event
     return new HttpDownloadValue(descriptor.directory);
   }
 
+  private static void setProxyIfNeeded() throws IOException {
+	if(!"".equalsIgnoreCase(System.getenv("HTTP_PROXY"))){
```

Comment:
I'll see what I can come up with on this.  Ideally proxy setup would be performed just once instead of for every resource requested.  That will involve more refactoring though and I'm hesitant to get muck with too much of you guys' stuff in here.